### PR TITLE
feat(WeightedSumGraphRepresentation): Allow clipping node reprs before summation

### PR DESCRIPTION
**Pull request by mmjb**
This implements a new feature to clip values of the (transformed) node representations before summing them up. This is particularly important for using the layer using the (weighted) summation modes ("sigmoid", "none"), as those otherwise can lead to values of extremely large magnitudes. [I've noticed by trying them out in the MoLeRVAE setting.] Clipping before summation allows the layer to still perform tasks such as counting the number of nodes, which would be limited by clipping after summation.

### CODE REVIEW

File: tf2_gnn/layers/nodes_to_graph_representation.py [hunk 1]
```diff
@@ -174,9 +186,13 @@ def call(self, inputs: NodesToGraphRepresentationInput, training: bool = False):
                 raise ValueError()
 
         # (2) compute representations for each node/head pair:
-        node_reprs = self._transformation_mlp(
-            inputs.node_embeddings, training=training
+        node_reprs = self._transformation_mlp_activation_fun(
```
Comment (pashminacameron) [inline tf2_gnn/layers/nodes_to_graph_representation.py:189]: Adding the missing activation function: We did this somewhere else very recently, how is this related (is it the base class)?

File: tf2_gnn/layers/nodes_to_graph_representation.py [hunk 2]
```diff
@@ -95,6 +97,12 @@ def __init__(
                 representations.
             transformation_mlp_dropout_rate: MLP inter-layer dropout rate for computing graph
                 representations.
+            transformation_mlp_result_lower_bound: Lower bound that results of the transformation
+                MLP will be clipped to before being scaled and summed up.
+                This is particularly useful to limit the magnitude of results when using "sigmoid"
+                or "none" as weighting function.
+            transformation_mlp_result_upper_bound: Lower bound that results of the transformation
+                MLP will be clipped to before being scaled and summed up.
```
Comment (kmaziarz) [inline tf2_gnn/layers/nodes_to_graph_representation.py:105]: typo: Lower -> Upper

File: tf2_gnn/layers/nodes_to_graph_representation.py [hunk 3]
```diff
@@ -74,6 +74,8 @@ def __init__(
         transformation_mlp_layers: List[int] = [128],
         transformation_mlp_activation_fun: str = "ReLU",
         transformation_mlp_dropout_rate: float = 0.2,
+        transformation_mlp_result_lower_bound: Optional[int] = None,
+        transformation_mlp_result_upper_bound: Optional[int] = None,
```
Comment (kmaziarz) [inline tf2_gnn/layers/nodes_to_graph_representation.py:78]: Why `int`s instead of `float`s?

File: tf2_gnn/layers/nodes_to_graph_representation.py (hunk 1 above)
Comment (pashminacameron) [reply tf2_gnn/layers/nodes_to_graph_representation.py:189]: Found it, it was in the padded_node_to_graph_representation layer, not in this repo. Not a base class, but very similar.

Comment (mmjb) [reply tf2_gnn/layers/nodes_to_graph_representation.py:189]: I didn't remember this at all, but you're right: We did this for `PaddedWeightedSumGraphRepresentation` in the other repo. That class essentially a code copy of this (but avoiding the complexity of using sparse operations). So conceptually related, but no formal relationship in the inheritance sense.

File: tf2_gnn/layers/nodes_to_graph_representation.py (hunk 2 above)
Comment (mmjb) [reply tf2_gnn/layers/nodes_to_graph_representation.py:105]: Thanks; fixed.

File: tf2_gnn/layers/nodes_to_graph_representation.py (hunk 3 above)
Comment (mmjb) [reply tf2_gnn/layers/nodes_to_graph_representation.py:78]: No good reason, just my brain being slow about it...
